client/js/Ui/searchpanel.js

```diff
@@ -1,89 +1,95 @@
+            const NODE = "_node";
+
+            const template = new sc.ScTemplate();
+            template.triple(
+                [sc.ScType.Unknown, NODE],
+                sc.ScType.EdgeDCommonVar,
+                linkAddr,
+                sc.ScType.EdgeAccessVarPosPerm,
+                identification,
+            );
+            let result = await window.scClient.templateSearch(template);
+            if (result.length) {
+                return result[0].get(NODE).value;
+            }
+
+            return null;
+        };
+
+        let linkAddrs = await window.scClient.getLinksByContents([string]);
+        let addr = null;
+
+        if (linkAddrs.length) {
+            linkAddrs = linkAddrs[0];
+
+            if (linkAddrs.length)
+            {
+                addr = linkAddrs[0];
+                addr = await searchNodeByIdentifier(addr, window.scKeynodes["nrel_system_identifier"]);
+                if (!addr) {
+                    addr = await searchNodeByIdentifier(addr, window.scKeynodes["nrel_main_idtf"]);
+                }
+
+                if (!addr) {
+                    addr = linkAddrs[0];
+                }
+            }
+
+            SCWeb.core.Main.doDefaultCommand([addr]);
+        }
     }
     event.stopPropagation();
     $('.typeahead').val('');
     $('.tt-dropdown-menu').hide();
 };
 
-const searchByIdentifier = (identifier) => {
-    SCWeb.core.Server.resolveScAddr([identifier]).then(function (addrs) {
-        SCWeb.core.Main.doDefaultCommand([addrs[identifier]]);
-    });
-}
-
 SCWeb.ui.SearchPanel = {
     init: function () {
         return new Promise(resolve => {
-            var keynode_nrel_main_idtf = null;
-            var keynode_nrel_idtf = null;
-            var keynode_nrel_system_idtf = null;
-
             $('.typeahead').typeahead({
                   minLength: 3,
                   highlight: true,
-              },
-              {
+            },
+            {
                   name: 'idtf',
-                  source: function (query, cb) {
+                  source: function (str, cb) {
                       $('#search-input').addClass('search-processing');
-                      SCWeb.core.Server.findIdentifiersSubStr(query, function (data) {
-                          keys = [];
+                      window.scClient.getLinksContentsByContentSubstrings([str]).then((result) => {
+                          let keys = [];
+
+                          if (result.length) {
+                              const maxContentSize = 30;
 
-                          var addValues = function (key) {
-                              var list = data[key];
-                              if (list) {
-                                  for (idx in list) {
-                                      var value = list[idx]
-                                      keys.push({name: value[1], addr: value[0], group: key});
+                              const slice = result[0];
```

Comment:
Found a bug: search by idtf not working in SCg
![image](https://user-images.githubusercontent.com/8840551/194565052-0ab491d5-9995-4cdc-bde7-2a54230720e3.png)